pkg/targetgroupbinding/resource_manager.go

```diff
@@ -382,8 +383,13 @@ func (m *defaultResourceManager) deregisterTargets(ctx context.Context, tgARN st
 	return m.targetsManager.DeregisterTargets(ctx, tgARN, sdkTargets)
 }
 
-func (m *defaultResourceManager) registerPodEndpoints(ctx context.Context, tgARN string, endpoints []backend.PodEndpoint) error {
-	vpcInfo, err := m.vpcInfoProvider.FetchVPCInfo(ctx, m.vpcID)
+func (m *defaultResourceManager) registerPodEndpoints(ctx context.Context, tgARN, tgVpcId string, endpoints []backend.PodEndpoint) error {
+	vpcId := m.vpcID
+	// Target group is in a different VPC from the cluster's VPC
+	if tgVpcId != "" && tgVpcId != m.vpcID {
+		vpcId = tgVpcId
```

Comment:
Probably better to call out via the logger that the specified VPC in TGB is different from cluster vpc.

Reply:
comment addressed